server-extension/src/main/java/flyway/olcesium/V1_1__set_layers.java

```diff
@@ -0,0 +1,212 @@
+package flyway.olcesium;
+
+import fi.nls.oskari.domain.map.OskariLayer;
+import fi.nls.oskari.domain.map.view.Bundle;
+import fi.nls.oskari.domain.map.view.View;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.map.layer.formatters.LayerJSONFormatterWMTS;
+import fi.nls.oskari.map.view.ViewService;
+import fi.nls.oskari.map.view.ViewServiceIbatisImpl;
+import fi.nls.oskari.service.OskariComponentManager;
+import fi.nls.oskari.service.capabilities.CapabilitiesCacheService;
+import fi.nls.oskari.service.capabilities.OskariLayerCapabilities;
+import fi.nls.oskari.util.JSONHelper;
+import fi.nls.oskari.wmts.WMTSCapabilitiesParser;
+import fi.nls.oskari.wmts.domain.WMTSCapabilities;
+import fi.nls.oskari.wmts.domain.WMTSCapabilitiesLayer;
+import org.flywaydb.core.api.migration.jdbc.JdbcMigration;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class V1_1__set_layers implements JdbcMigration {
+
+    private static final Logger LOG = LogFactory.getLogger(flyway.olcesium.V1_1__set_layers.class);
+    private static final String CESIUM_VIEW_NAME = "Geoportal Ol Cesium";
+    private static final String MAP_BUNDLE_NAME = "mapfull";
+    private static final String BASELAYER_PLUGIN_NAME = "BackgroundLayerSelectionPlugin";
+
+    private ViewService viewService = null;
+    private CapabilitiesCacheService capabilitiesService = null;
+
+    public void migrate(Connection connection) throws SQLException {
+        viewService =  new ViewServiceIbatisImpl();
+        capabilitiesService = OskariComponentManager.getComponentOfType(CapabilitiesCacheService.class);
+        List<OskariLayer> layers = getNlsBaseLayers(connection);
+        LOG.info("Start populating matrixies for Oskari WMTS layers - count:", layers.size());
+        for (OskariLayer layer : layers) {
+            updateNlsBaseLayer(connection, layer);
+        }
+        if (!layers.isEmpty()) {
+            updateCesiumViews(connection, layers);
+        }
+    }
+
+    private void updateNlsBaseLayer(Connection connection, OskariLayer layer) {
+        try {
+            // update capabilities in cache
+            String data = CapabilitiesCacheService.getFromService(layer);
+            OskariLayerCapabilities caps = new OskariLayerCapabilities(
+                    layer.getSimplifiedUrl(true),
+                    layer.getType(),
+                    layer.getVersion(),
+                    data);
+            capabilitiesService.save(caps);
```

Comment:
This should be moved down a bit. We don't want to save the XML to the DB if WMTSCapabilitiesParser fails to parse it.